### DISCUSSION TITLE
set up renovate for dependency updates

### DISCUSSION THREAD

**Discussion by boredland** [taskforcesh/bullmq#1348] [Ideas]
as bullmq has quite some dependencies and a pretty good test coverage, I think having some kind of automation around dependency upgrades would be nice. In my experience the "grouping" feature of renovate is a game changer (compared to dependabot) when it comes to noise reduction. Wdyt?

**Comment by manast** [comment]
Regarding upgrading dependencies I am still a bit conservative about it, so I am not fully comfortable with automatic upgrade just yet. Also we do not have so many direct dependencies, it is mostly the dependency to IORedis, which we upgrade carefully as any issue with it could lead to problems in BullMQ. Of course we get sometimes security related upgrades, but they are usually quite minor.

**Reply by boredland** [reply]
valid, but wouldn't then disabling auto-merge and [disabling prs for devDependencies](https://docs.renovatebot.com/presets-default/#disabledevdependencies) already make sense to stay informed and "have a PR open" for the direct dependencies?

**Reply by manast** [reply]
yeah, maybe it is a good idea.

**Reply by boredland** [reply]
added here: https://github.com/taskforcesh/bullmq/pull/1346/commits/240332af54ffb61e34319bb87abe6ef7ddabf6a1
